lib/streamlit/server/Server.py

```diff
@@ -116,8 +118,34 @@ def start(self, on_started):
 
         LOGGER.debug("Starting server...")
         app = self._create_app()
+
+        def start_listening(call_count=0):
+            port = config.get_option("server.port")
+            try:
+                app.listen(port)
+            except OSError as e:
+                ADDRESS_ALREADY_IN_USE_ERROR_CODE = 48
```

Comment:
Move this to top of file.

Reply:
updated this to use `errno` that's coming right from the C lib bindings.